docs/DeveloperGuide.adoc

```diff
@@ -278,7 +278,88 @@ are applied when restoring `**THRIFT**` data from the data file.
 ** Pros: Calculations become trivial: to get balance, sum up the entire `**List<Transaction>**`.
 ** Cons: Overhead to parse user input to allow only a single negative symbol and no other symbols.
 
-=== Tagging/Untagging transactions
+// tag::taguntag[]
+=== Tag/Untag feature
+==== Implementation
+
+The **tag**/**untag** command performs modifications on existing `**Transaction**` entries. Due to polymorphism, both **Expense**` and `**Income**` objects extends the abstract class `**Transaction**`, and are treated the same way in the context of the **tag**/**untag** command.
+
+.A Class Diagram illustrating the implementation of Tags inside a Transaction
+image::tag-untag/Polymorphism.png[]
+
+Because of polymorphism, many of the driver functions in `THRIFT` uses `Transaction` for both `Expense`
+and `Income` objects. For example, when tagging a new `Expense`/`Income`, the `TagCommandParser` will treat them as the same object:
+
+image::tag-untag/TagSequence.png[]
+
+When a user enters a **tag**/**untag** command, it is parsed by the respective `TagCommandParser#parse(String args)` and `UntagCommandParser#parse(String args)` parsers and returns a `**TagCommand**` command object or a `**UntagCommand**` command object  respectively, which will be executed. For both commands, the following conditions will cause a `**ParseException**` to be thrown by their respective parsers:
+
+. Missing parameters
+. Incorrect syntax (i.e. missing prefix, if it is required)
+. Illegal values in parameters (i.e. special characters and symbols entered for an integer-only field)
+. Multiple occurrences of parameters which only expects a single entry
+
+[NOTE]
+If the user input is incorrect due to any of the reasons above, the corresponding usage syntax will be shown.
+
+The following is an example on how a `**Tag**` is added and removed from a `**Transaction**`, with details on the processes in the backend.
+
+Step 1. The user launches the application with data from previous sessions. THRIFT does contains only one `**Transaction**` containing no tags and the user is going to perform tagging and un-tagging operations on it.
+
+image::tag-untag/TagUntagTransactionWoTag.png[]
+
+Step 2. The user executes the command **tag i/1 t/Delicious** to tag the `**Transaction**` at `**Index**` 1 with the `**Tag**` "Delicious". The input is checked with a `**TagCommandParser**` parser and an attempt to parse each parameter occurs:
+
+* `**Index**` is parsed by `ParserUtil#parseIndex(String)`
+* `**Tag**` strings are parsed by `ParserUtil#parseTags(Collection<String>)` which calls `ParserUtil#parseTag(String)` iteratively for every string in the collection
+
+TIP: `**ParserUtil**` is a class that contains useful functions for parsing the inputs from the user
```

Comment:
Should add full-stop at the end since it sounds like a complete sentence.

Reply:
Full-stop has been added.